File: 0019 - Remove Nth Node From End of List - Medium.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    struct ListNode *ptr1 = malloc(sizeof(struct ListNode));
    ptr1->next = head;

    struct ListNode *f = ptr1;
    struct ListNode *s = ptr1;

    for(int i = 0; i<n+1; i++){
        f = f->next;
    }

    while(f!=NULL){
        f = f->next;
        s = s->next;
    }
    s->next = s->next->next;

    return ptr1->next;
}

File: 0019 - Remove Nth Node From End of List - Medium.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(0, head);
        ListNode curr = head;

        int count = 0;
        while (curr != null) {
            count++;
            curr = curr.next;
        }
        
        int k = count - n;

        ListNode kth = dummy;
        for (int i = 0; i < k; i++) {
            kth = kth.next;
        }
        kth.next = kth.next.next;
        return dummy.next;
    }
}
